machinery-helper/Rakefile

```diff
@@ -15,90 +15,23 @@
 # To contact SUSE about this file by physical or electronic mail,
 # you may find current contact information at www.suse.com
 
+require_relative "../tools/helper_rake_methods"
+
 task :default => "try_build"
 
 HELPER_DIR = File.expand_path(File.dirname(__FILE__))
-GIT_REVISION_FILE = File.join(HELPER_DIR, "..", ".git_revision")
-
-def write_go_version_file
-  file = <<-EOF
-// This is a generated file and shouldn't be changed
-
-package main
-
-const VERSION = "#{git_revision}"
-  EOF
-  File.write(File.join(HELPER_DIR, "version.go"), file)
-end
-
-def build_machinery_helper
-  FileUtils.rm_f(File.join(HELPER_DIR, "machinery-helper"))
-  Dir.chdir(HELPER_DIR) do
-    puts("Building machinery-helper binary.")
-    if !system("go build")
-      STDERR.puts("Warning: Building of the machinery-helper failed!")
-      false
-    else
-      true
-    end
-  end
-end
-
-def go_available?
-  if !system("which go > /dev/null 2>&1")
-    STDERR.puts(
-      "Warning: The Go compiler is not available on this system. Skipping building the" \
-        " machinery-helper.\nThe machinery-helper increases the inspection speed significantly."
-    )
-    false
-  else
-    true
-  end
-end
-
-def arch_supported?
-  arch = `uname -p`.chomp
-  if !["x86_64"].include?(arch)
-    STDERR.puts(
-      "Warning: The hardware architecture #{arch} is not yet supported by the machinery-helper."
-    )
-    false
-  else
-    true
-  end
-end
-
-def runs_in_git?
-  Dir.exist?(File.join(HELPER_DIR, "..", ".git")) && system("which git > /dev/null 2>&1")
-end
-
-def git_revision
-  `git rev-parse HEAD`.chomp
-end
-
-def write_git_revision_file
-  File.write(GIT_REVISION_FILE, git_revision)
-end
-
-def changed_revision?
-  if File.exist?(GIT_REVISION_FILE)
-    old_revision = File.read(GIT_REVISION_FILE)
-  else
-    old_revision = "unknown"
-  end
-  git_revision != old_revision
-end
 
 def run_build
+  helper_build = HelperBlueprint.new(HELPER_DIR)
```

Comment:
I don't get the naming here. What is the responsibility of `HelperBlueprint` and why is the variable called `helper_build`?